android/src/main/java/expo/modules/xmtpreactnativesdk/XMTPModule.kt

```diff
@@ -144,23 +151,29 @@ class XMTPModule : Module() {
             }
         }
 
-        AsyncFunction("loadMessages") { clientAddress: String, conversationTopic: String, conversationID: String?, limit: Int?, before: Long?, after: Long? ->
-            val conversation =
+        AsyncFunction("loadMessages") { clientAddress: String, topics: List<String>, conversationIDs: List<String?>, limit: Int?, before: Long?, after: Long? ->
+            logV( "loadMessages");
+            // TODO: use batchQuery instead of one-at-a-time (once Android SDK supports it).
+            val beforeDate = if (before != null) Date(before) else null
+            val afterDate = if (after != null) Date(after) else null
+            topics.zip(conversationIDs) { topic, conversationID ->
                 findConversation(
                     clientAddress = clientAddress,
-                    topic = conversationTopic,
+                    topic = topic,
                     conversationId = conversationID
                 )
-                    ?: throw XMTPException("no conversation found for $conversationTopic")
-            val beforeDate = if (before != null) Date(before) else null
-            val afterDate = if (after != null) Date(after) else null
-
-            conversation.messages(limit = limit, before = beforeDate, after = afterDate)
-                .map { DecodedMessageWrapper.encode(it) }
+                    ?: throw XMTPException("no conversation found for $topic")
+            }
+                .map {
+                    it.messages(limit = limit, before = beforeDate, after = afterDate)
+                        .map { DecodedMessageWrapper.encode(it) }
+                }
+                .flatten()
```

Comment:
You can actually do this with a `flatMap` in Kotlin https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/flat-map.html